examples/turnip-town/move/sources/field.move

```diff
@@ -0,0 +1,547 @@
+/// # Field
+///
+/// Defines the `Field` type, which is responsible for one instance of
+/// the game.  `Field`s are stored centrally, in a `Game` instance,
+/// and a `Deed` is created that gives its owner access to that
+/// `Field` through the `Game`.
+///
+/// Although `Field` is not an object, an `ID` is generated for it, to
+/// identify it in the `Game`.  This `ID` is also stored in the `Deed`.
+///
+/// Any field owner can sow or harvest turnips from the field, but
+/// simulating growth and watering can only be done via the `Game`
+/// which decides the cadence of simulation and how much water to
+/// dispense.
+module turnip_town::field {
+    use std::option::{Self, Option};
+    use std::vector;
+    use sui::math;
+    use sui::object::{Self, ID, UID};
+    use sui::tx_context::TxContext;
+
+    use turnip_town::turnip::{Self, Turnip};
+
+    friend turnip_town::game;
```

Comment:
I ended up following a pattern where there's a sequence of cascading friends: `game` is `field`'s friend, and `field` is `turnip`'s friend.  This was so that `game` could perform privileged operations on `field` and same for `field` on `turnip`.  I couldn't come up with a great way to avoid that without putting everything in one big file, but suggestions welcome.